### PR TITLE
Added NPI validity check and practitioner indentifier to FHIR bundle

### LINKED ISSUE #5618: Add validated NPIs to FHIR messages
### Background
We previously were not including provider NPIs because many of ours are invalid. CalREDIE needs the NPI field completed as accurately as possible, so we need to add it to our bundles. They have asked us to include the value IF it is valid.

### Action requested
When building a FHIR message from a test event, conditionally include the NPI for the provider in Practitioner.identifier of our messages. ~The conditions to include the NPI should be that it is an NPI for a valid provider returned by the [NPI API](https://clinicaltables.nlm.nih.gov/apidoc/npi_org/v3/doc.html).~ The condition to include the NPI should be that it is a 10-digit number that passes the Luhn algorithm check.

### Acceptance Criteria
* FHIR messages generated for a test events at a facility with a provider with a valid NPI should include the NPI in Practitioner.identifier
  * Practitioner.identifier.system should be `http://hl7.org/fhir/sid/us-npi` 
* messages generated for which there is not a valid NPI should not have the Practitioner.identifier

### Additional context
NPI Luhn check validation: https://npiprofile.com/validation/
NPI constraints for US Core Practitioners: https://build.fhir.org/ig/HL7/US-Core/StructureDefinition-us-core-practitioner.html#constraints

https://commons.apache.org/proper/commons-validator/apidocs/org/apache/commons/validator/routines/checkdigit/LuhnCheckDigit.html
https://docs.jboss.org/hibernate/validator/5.1/api/org/hibernate/validator/constraints/LuhnCheck.html

### PR DESCRIPTION
**Pull request by johanna-skylight**
# BACKEND PULL REQUEST

## Related Issue

- Resolves #5618 

## Changes Proposed

- Add the apache library [commons-validator](https://commons.apache.org/proper/commons-validator/apidocs/org/apache/commons/validator/routines/checkdigit/LuhnCheckDigit.html) to the project
- Include logic in the convertPractitioner method that checks if the NPI info is valid and add this data into the bundle if that is the case.
- Add necessary test cases to validate the bundle gets properly created if the npi info is null, blank or invalid.

## Additional Information
The validation for the 10 digit number NPI does not really map to the luhn algorithm 1:1 unless the NPI prefix is added to the NPI code for the check.

So for this implementation I added the NPI prefix to the NPI value and ran the Luhn check with that. For more information on where I got the prefix please check the validation steps of a NPI from the [CMS docs](https://www.cms.gov/Regulations-and-Guidance/Administrative-Simplification/NationalProvIdentStand/Downloads/NPIcheckdigit.pdf).

See below an explanation on how the validation is done without the prefix. I did not follow this path as it would need more custom code.
Screenshot 2023-04-12 at 3 59 15 PM

## Testing
_This code is deployed in dev2_
- Set a valid NPI to the facility you will be testing with (use [this tool](https://npiprofile.com/clia) to get valid NPI)
- Conduct a test for that facility and verify the bundle contains the NPI info as part of the practitioner entry.

### CODE REVIEW

**Review by BobanL — APPROVED** [review]
Changes look good! Thanks for adding this 😄

File: backend/src/test/resources/fhir/practitionerNPI.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "resourceType" : "Practitioner",
+  "id": "3c9c7370-e2e3-49ad-bb7a-f6005f41cf29",
+  "name" : [ {
+    "family" : "Earhart",
+    "given" : [ "Amelia", "Mary" ]
```
Comment (emyl3) [inline backend/src/test/resources/fhir/practitionerNPI.json:6]: ❤️ aww amelia!

**Review by emyl3 — APPROVED** [review]
LGTM! Saw the new identifier value for the practitioner in dev2 👍
